x-pack/plugins/apm/server/routes/service_groups/get_service_group_alerts.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kqlQuery } from '@kbn/observability-plugin/server';
+import {
+  AggregationsCardinalityAggregate,
+  AggregationsFilterAggregate,
+  QueryDslQueryContainer,
+} from '@elastic/elasticsearch/lib/api/types';
+import { ApmPluginRequestHandlerContext } from '../typings';
+import { SavedServiceGroup } from '../../../common/service_groups';
+
+export async function getServiceGroupAlerts({
+  serviceGroups,
+  authorizedAlertsIndices,
+  context,
+}: {
+  serviceGroups: SavedServiceGroup[];
+  authorizedAlertsIndices?: string[];
+  context: ApmPluginRequestHandlerContext;
+}) {
+  if (
+    !authorizedAlertsIndices ||
+    authorizedAlertsIndices.length === 0 ||
+    serviceGroups.length === 0
+  ) {
+    return {};
+  }
+  const serviceGroupsKueryMap: Record<string, QueryDslQueryContainer> =
+    serviceGroups.reduce((acc, sg) => {
+      return {
+        ...acc,
+        [sg.id]: kqlQuery(sg.kuery)[0],
+      };
+    }, {});
+
+  const esClient = (await context.core).elasticsearch.client;
+  const params = {
+    index: authorizedAlertsIndices,
+    size: 0,
+    query: {
+      bool: {
+        filter: [
+          { term: { 'kibana.alert.rule.producer': 'apm' } },
+          { term: { 'kibana.alert.status': 'active' } },
+        ],
+      },
+    },
+    aggs: {
+      service_groups: {
+        filters: {
+          filters: serviceGroupsKueryMap,
+        },
+        aggs: {
+          alerts_count: {
+            cardinality: {
+              field: 'kibana.alert.uuid',
+            },
+          },
+        },
+      },
+    },
+  };
+  const result = await esClient.asCurrentUser.search(params);
```

Comment:
How does this work? These are supposed to be system indices that are not accessible to the end user. Have you tried this with non-superusers?

Reply:
Right now if the user doesn't have access to query the internal index, it just omits the alert counts from the response. I can look into the rule Data client, if it already handles this, or we could use the internal user as a last resort. 
I also discovered a related issue that Service groups breaks down totally if the user doesn't have kibana read privileges for the 'Saved objects management' feature.

Reply:
I don't think this is how those indices should be accessed. They should be accessed using the internal user, and there should be an evaluation before that whether the user has access to the alerts feature. Non-superusers will not have access to these indices (and perhaps even superusers won't, I remember there was a recent change wrt system indices). I don't think omitting the alert count in many cases is a good workaround. Can you reach out to the alerting teams to figure out how we should handle it if we want to read from the alerts indices?